From e5e9af7ee4f34c0db2b3a0f7d60c708c6c10bbae Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Sat, 12 Aug 2017 12:08:45 +1000
Subject: [PATCH 15/16] Enable and make BFQ default IO scheduler, overriding
 inherited default_cfq value and enable default kyber scheduler option.

---
 block/Kconfig.iosched | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/block/Kconfig.iosched b/block/Kconfig.iosched
index fd2cefa47d35..92a5082af1da 100644
--- a/block/Kconfig.iosched
+++ b/block/Kconfig.iosched
@@ -42,7 +42,7 @@ config CFQ_GROUP_IOSCHED
 choice
 
 	prompt "Default I/O scheduler"
-	default DEFAULT_CFQ
+	default DEFAULT_BFQ
 	help
 	  Select the I/O scheduler which will be used by default for all
 	  block devices.
@@ -50,9 +50,15 @@ choice
 	config DEFAULT_DEADLINE
 		bool "Deadline" if IOSCHED_DEADLINE=y
 
-	config DEFAULT_CFQ
+	config DEFAULT_BFQ
+		bool "BFQ" if IOSCHED_BFQ=y
+
+	config DEFAULT_CFQ_NOINHERIT
 		bool "CFQ" if IOSCHED_CFQ=y
 
+	config DEFAULT_KYBER
+		bool "KYBER" if MQ_IOSCHED_KYBER=y
+
 	config DEFAULT_NOOP
 		bool "No-op"
 
@@ -61,7 +67,9 @@ endchoice
 config DEFAULT_IOSCHED
 	string
 	default "deadline" if DEFAULT_DEADLINE
-	default "cfq" if DEFAULT_CFQ
+	default "bfq" if DEFAULT_BFQ
+	default "cfq" if DEFAULT_CFQ_NOINHERIT
+	default "kyber" if DEFAULT_KYBER
 	default "noop" if DEFAULT_NOOP
 
 config MQ_IOSCHED_DEADLINE
@@ -81,7 +89,7 @@ config MQ_IOSCHED_KYBER
 
 config IOSCHED_BFQ
 	tristate "BFQ I/O scheduler"
-	default n
+	default y
 	---help---
 	BFQ I/O scheduler for BLK-MQ. BFQ distributes the bandwidth of
 	of the device among all processes according to their weights,
-- 
2.11.0

